### ISSUE TITLE
[NEWS] GPSR - GET /sale/products - dodaliśmy dane o bezpieczeństwie produktu / GPSR - GET /sale/products - we have added product safety data

### ISSUE THREAD

**Issue by MaciejFrackowiak** [allegro/allegro-api#10483]
**13 grudnia 2024** zacznie obowiązywać nowe rozporządzenie Unii Europejskiej **[General Product Safety Regulation](https://commission.europa.eu/business-economy-euro/doing-business-eu/eu-product-safety-and-labelling/product-safety/general-product-safety-regulation_en)**. Celem GPSR jest pokazanie kupującemu, skąd pochodzi produkt, który może kupić na Allegro.

**Jak było dotychczas?**

Jeżeli wybrany produkt z Katalogu Produktów Allegro posiada zdefiniowane dane o bezpieczeństwie produktu (“**productSafety**”), które zawierają dane teleadresowe producenta (“**responsibleProducers**”) oraz informacje o bezpieczeństwie produktu (“**safetyInformation**”), zwracaliśmy je wyłącznie w odpowiedzi do [GET /sale/products/{productId}](https://developer.allegro.pl/documentation/#operation/getSaleProduct).

**Co zmieniliśmy?**

Dzisiaj, dla [GET /sale/products](https://developer.allegro.pl/documentation/#operation/getSaleProducts), udostępniliśmy również pole “**productSafety**”, dzięki któremu już w wynikach wyszukiwania produktów zwracamy dane teleadresowe producenta (“**responsibleProducers**”) oraz informacje o bezpieczeństwie produktu (“**safetyInformation**”), jeżeli wybrany produkt z Katalogu Produktów Allegro je posiada.

**Przykładowy response:**

```json5
{
	"products": [
    	   {
        	"id": "f5c67f19-851e-4bb3-bbfb-2d873ac3e2e8",
        	"name": "ŻARÓWKA 8GH 002 089-361 HELLA",
		...
        	"productSafety": {
                   "responsibleProducers": [
                	{
                          "id": "e1000ecb-a77d-45f6-97ca-b03c484de58e",
                          "name": "moto - Hella",
                          "producerData": {
                        	"tradeName": "Hella",
                        	"address": {
                            	   "street": "Rixbecker Str.75",
                            	   "postalCode": "59552",
                            	   "city": "Lippstadt",
                            	   "countryCode": "DE"
                        	},
                        	"contact": {
                            	   "email": "info@hella.com",
                            	   "phoneNumber": null
                        	}
                          }
                	}
             	   ],
            	   "safetyInformation": {
                	"type": "ATTACHMENTS",
                	"attachments": [
                          {
                        	"id": "34b2f604-c585-4597-8c35-2446073c845c"
                          }
                	]
            	    }
        	}
    	   }
	],
	...
	"filters": [],
	"nextPage": null
}
```
> 

Tym samym, przekazując identyfikator produktu (“**product.id**”) podczas tworzenia oferty ([POST /sale/product-offers](https://developer.allegro.pl/documentation/#operation/createProductOffers)), uwzględnimy dane o bezpieczeństwie produktu w samej ofercie. 

**Dlaczego wprowadziliśmy zmianę?**

Zależy nam, aby sprzedający mogli w możliwie najwygodniejszy dla siebie sposób uzupełnić oferty danymi teleadresowymi producentów oraz informacjami o bezpieczeństwie produktów, które z dniem **13 grudnia 2024** będą wymagane zgodnie z rozporządzeniem GPSR.

-----

On **December 13, 2024**, the new European Union Regulation **[General Product Safety Regulation](https://commission.europa.eu/business-economy-euro/doing-business-eu/eu-product-safety-and-labelling/product-safety/general-product-safety-regulation_en)** will come into effect. The goal of the GPSR is to show the buyer where the product available on Allegro originates.

**How was it so far?**

If the selected product from the Allegro Product Catalog has the defined product safety data (“**productSafety**”) which contains the contact information of the product producer (“**responsibleProducers**”) and product safety information (“**safetyInformation**”), we returned it only in the response to [GET /sale/products/{productId}](https://developer.allegro.pl/documentation/#operation/getSaleProduct).

**What have we changed?**

Today, for [GET /sale/products](https://developer.allegro.pl/documentation/#operation/getSaleProducts), we have also provided the “**productSafety**” field, which already returns the contact information of the product producer (“**responsibleProducers**”) and product safety information (“**safetyInformation**”) in the product search results, if the selected product from the Allegro Product Catalog has it.

**Sample response:**

```json5
{
	"products": [
    	   {
        	"id": "f5c67f19-851e-4bb3-bbfb-2d873ac3e2e8",
        	"name": "ŻARÓWKA 8GH 002 089-361 HELLA",
		...
        	"productSafety": {
                   "responsibleProducers": [
                	{
                          "id": "e1000ecb-a77d-45f6-97ca-b03c484de58e",
                          "name": "moto - Hella",
                          "producerData": {
                        	"tradeName": "Hella",
                        	"address": {
                            	   "street": "Rixbecker Str.75",
                            	   "postalCode": "59552",
                            	   "city": "Lippstadt",
                            	   "countryCode": "DE"
                        	},
                        	"contact": {
                            	   "email": "info@hella.com",
                            	   "phoneNumber": null
                        	}
                          }
                	}
             	   ],
            	   "safetyInformation": {
                	"type": "ATTACHMENTS",
                	"attachments": [
                          {
                        	"id": "34b2f604-c585-4597-8c35-2446073c845c"
                          }
                	]
            	    }
        	}
    	   }
	],
	...
	"filters": [],
	"nextPage": null
}
```
> 

Thus, by passing the product identifier (“**product.id**”) when creating an offer ([POST /sale/product-offers](https://developer.allegro.pl/documentation/#operation/createProductOffers)), we will include the product safety data in the offer itself.

**Why have we made the change?**

We want sellers to be able to complete their offers as conveniently as possible with the contact information of product producers and product safety information, which will be required under the GPSR regulation as of **December 13, 2024**.

**Comment by morawcik** [comment]
Jeśli chodzi o załączniki w safetyInformation - w jakim celu są zwracane? Tzn można je jakoś wykorzystać (do wglądu czy coś) czy to na zasadzie "tutaj jest jakiś załącznik" i koniec?

**Comment by MaciejFrackowiak** [comment]
@morawcik To są dane dostarczone przez producenta/dystrybutora, etc. Możesz te dane przypisać do swoich produktów.

**Comment by morawcik** [comment]
No tak tylko, że po prostu przypisując ofercie id produktu te dane się przypisują automatycznie, więc do tego nie są potrzebne zwracane te dane. I o ile jeszcze dane producenta można wykorzystać do ich zweryfikowania tak już z załącznikami nic nie zrobimy a i tak automatycznie są przekazywane do oferty. Stąd moja ciekawość czy jest jakaś możliwość ich podejrzenia/weryfikacji.

**Comment by MaciejFrackowiak** [comment]
@morawcik Docelowo zostanie udostępniony endpoint pozwalający na pobranie szczegółów załącznika. Oczywiście zawsze można zdefiniować własne dane, bez korzystania z sugerowanych danych.

**Comment by morawcik** [comment]
Super, dziękuję za informacje :)

**Comment by creativepl-php** [comment]
Fajna zmiana. Rozumiem, że jeśli mam książkę "W pustyni i w puszczy" o numerze ISBN 123456789, od wydawnictwa "Prószyński i S-ka" i dodam do tego produktu dane teleadresowe wydawnictwa (producenta), to przy kolejnym wystawianiu tej samej książki, nie będę już musiał tych danych podawać. Ale co z każdą kolejną książką wydawnictwa "Prószyński i S-ka"? Będzie trzeba za każdym razem te dane wybierać czy będą się ustawiać automatycznie?

**Comment by MaciejFrackowiak** [comment]
@creativepl-php Odpowiadając:
1. Nie, nie przypiszemy ich na stałe do produktu. Musisz zadbać o to we własnym zakresie.
2. To są dane od dystrybutora, które zostały dodane do wskazanego produktu - mają jednak charakter poglądowy, ponieważ to sprzedający decyduje, czy są one zgodne i czy chce je wykorzystać.
3. Nie, identyfikatory są stricte per konto użytkownika. Wyjątek - identyfikatory zwracane przez GET /sale/products / GET /sale/products/{productId} można wykorzystać dowolnie.
4. Od 13 grudnia 2024 obowiązuje rozporządzenie UE odnośnie GPSR - i co do zasady te dane prawnie są wymagane. My jednak nie oznaczyliśmy jeszcze tych danych jako obowiązkowe by nie "popsuć" integracji, które nie przygotowały jeszcze rozwiązań. Nie oznacza to jednak, że nasz brak obowiązkowości to zgoda na brak danych, które są prawnie wymagane - pola w każdym momencie mogą zostać oznaczone jako wymagane.

**Comment by creativepl-php** [comment]
@MaciejFrackowiak Dziekuje za odpowiedź. Czy planujecie wprowadzić jakieś oznaczenie dla parametrów, który parametr oznacza producenta? 
1) w przypadku książek będzie to parametr Wydawnictwo,
2) w przypadku telefonów będzie to Marka telefonu,
3) w przypadku lusterka (zwykłego, dekoracyjnego) będzie to Marka,
4) w przypadku części samochodowych będzie to Producent części,
5) w przypadku obrazu będzie to Marka
6) w przypadku przywołanej żarówki Hella będzie to Producent.
Namnożyły się te parametry i nie wiadomo jak to wiązać.

**Comment by MaciejFrackowiak** [comment]
@creativepl-php Nie, nie ma takich planów. Odnośnie filtra - proszę dodaj swój pomysł jako [sugestię](https://github.com/allegro/allegro-api/discussions).
